File: PosHubApi/Dtos/AddressDto.cs
using System.Text.Json.Serialization;

namespace PosHubApi.Dtos
{
    public class AddressDto
    {
        [JsonPropertyName("address")]
        public string Address { get; set; }

        [JsonPropertyName("googlePlaceId")]
        public string GooglePlaceId { get; set; }

        [JsonPropertyName("latitude")]
        public double Latitude { get; set; }

        [JsonPropertyName("longitude")]
        public double Longitude { get; set; }

        [JsonPropertyName("addressComponents")]
        public AddressComponentsDto AddressComponents { get; set; }
    }
}

File: poshub-spa/src/app/dtos/addressDto.ts
import { AddressComponentsDto } from "./addressComponentsDto";

export class AddressDto {
  constructor(
    public address: string = '',
    public googlePlaceId: string = '',
    public latitude: number = 0,
    public longitude: number = 0,
    public addressComponents: AddressComponentsDto = new AddressComponentsDto(),
  ) { }
}
